src/genericitems/serialisers.cpp

```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2021 Team Dissolve and contributors
+
+#include "genericitems/serialisers.h"
+#include "base/lineparser.h"
+#include "classes/neutronweights.h"
+#include "classes/partialset.h"
+#include "classes/xrayweights.h"
+#include "math/data1d.h"
+#include "math/data2d.h"
+#include "math/data3d.h"
+#include "math/histogram1d.h"
+#include "math/histogram2d.h"
+#include "math/histogram3d.h"
+
+GenericItemSerialiser::GenericItemSerialiser()
+{
+    // PODs
+    registerSerialiser<bool>([](const std::any &a, LineParser &parser, const CoreData &coreData) {
+        return parser.writeLineF("{}\n", DissolveSys::btoa(std::any_cast<bool>(a)));
+    });
+    registerSerialiser<double>([](const std::any &a, LineParser &parser, const CoreData &coreData) {
+        return parser.writeLineF("{}\n", std::any_cast<double>(a));
+    });
+    registerSerialiser<int>([](const std::any &a, LineParser &parser, const CoreData &coreData) {
+        return parser.writeLineF("{}\n", std::any_cast<int>(a));
+    });
+
+    // stdlib
+    registerSerialiser<std::string>([](const std::any &a, LineParser &parser, const CoreData &coreData) {
+        return parser.writeLineF("{}\n", std::any_cast<std::string>(a));
+    });
+    registerSerialiser<std::streampos>([](const std::any &a, LineParser &parser, const CoreData &coreData) {
+        return parser.writeLineF("{}\n", std::any_cast<std::streampos>(a));
+    });
+    registerSerialiser<std::vector<double>>([](const std::any &a, LineParser &parser, const CoreData &coreData) {
+        const auto &v = std::any_cast<const std::vector<double> &>(a);
+        if (!parser.writeLineF("{}\n", v.size()))
+            return false;
+        for (auto &n : v)
+            if (!parser.writeLineF("{:16.9e}\n", n))
+                return false;
+        return true;
+    });
```

Comment:
```suggestion
    registerSerialiser<std::vector<double>>([](const std::any &a, LineParser &parser, const CoreData &coreData) {
        const auto &v = std::any_cast<const std::vector<double> &>(a);
        if (!parser.writeLineF("{}\n", v.size()))
            return false;
        for (auto &n : v)
            if (!serialiseObject(n, parser, coreData))
                return false;
        return true;
    });
```

Just as with the deserialisers, I think it could be useful to go slightly recursive here.  I've mentioned before that we already have an inconsistency, plus this would be more useful on some of the later classes (example to follow).